scripts/validate-schemas-final.js

```diff
@@ -0,0 +1,71 @@
+const fs = require('fs');
+const path = require('path');
+
+
+function validation (excludedFiles){
+
+  // Specify the path to the 'schemas' directory
+  const directoryPath = './schemas';
+
+
+  const files = fs.readdirSync(directoryPath);
+
+  // Filter files
+  const filteredFiles = files.filter(file => !excludedFiles.includes(file));
+
+
+  // Iterate through the filtered files
+  filteredFiles.forEach(file => {
+    // Construct the full path to the JSON schema file
+    const filePath = path.join(directoryPath, file);
+
+    try {
+      // Read and parse the JSON schema
+      const fileContent = fs.readFileSync(filePath, 'utf8');
+      const obj = JSON.parse(fileContent);
+
+        if(obj.$schema === 'http://json-schema.org/draft-04/schema'){
+
+            // console.log('draft-04'); // for debugging purpose
+            const Ajv = require('ajv-draft-04');
+            const ajv = new Ajv();
+            // Validate the schema
+            const validate = ajv.validateSchema(obj);
+
+            // Check if the schema is valid
+            if (validate) {
+                console.log(`${file}: JSON Schema is valid!`);
+            } else {
+                console.error(`${file}: JSON Schema is not valid:`, ajv.errors);
+                process.exit(1);
+            }
+        } else{
+            // console.log('draft-07'); // for debugging purpose
+
+            const Ajv = require('ajv');
+            const ajv = new Ajv();
+
+            // Remove unnecessary definitions
+            delete obj.definitions['http://json-schema.org/draft-04/schema'];
+            delete obj.definitions['http://json-schema.org/draft-07/schema'];
+            // Validate the schema
+            const validate = ajv.validateSchema(obj);
+
+            // Check if the schema is valid
+            if (validate) {
+                console.log(`${file}: JSON Schema is valid!`);
+            } else {
+                console.error(`${file}: JSON Schema is not valid:`, ajv.errors);
+                process.exit(1);
```

Comment:
The issue of exiting here is that you are early-terminating the script once it founds the first error. In case there are more than one invalid JSON Schema file, you will only see the very first.

This can be specially annoying specially when running this script during CI, because you might be fixing one schema, and then waiting for another CI run to pass and find out that you have another invalid schema and so on.

I would suggest you collect errors in an array or whatever, and in the very end of the script you check if errors exist, and if so, print and exit(1).

Reply:
> The issue of exiting here is that you are early-terminating the script once it founds the first error. In case there are more than one invalid JSON Schema file, you will only see the very first.
>
> This can be specially annoying specially when running this script during CI, because you might be fixing one schema, and then waiting for another CI run to pass and find out that you have another invalid schema and so on.

Okay, I see...I get your point. :+1: 

> Do you believe you could change your code so you could validate all files and only then print all errors at the end of the execution?

Yes, I'll look into it. 🫡

Reply:
![image](https://github.com/asyncapi/spec-json-schemas/assets/99868037/6df24c05-33f4-4430-a70c-5213483f362d)
I guess this is what you suggested... :eyes: 

I've updated the code according to this. Please have a look
Ref: 17fcffdf3bab90fc807a92808eb0173467a0a209